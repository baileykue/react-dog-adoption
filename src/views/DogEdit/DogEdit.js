import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { fetchById } from '../../services/dogs';
import DogForm from '../../components/DogForm/DogForm';

export default function DogEdit() {
  const [dog, setDog] = useState({});
  const [loading, setLoading] = useState(true);
  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchById(params.id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [params.id]);

  const updateDog = (key, value) => {
    dog[key] = value;
    //this makes a NEW object, effectively updating each key value pair within the dog object
    setDog({ ...dog });
  };

  return (
    <div>
      <DogForm {...dog} updateDog={updateDog} />
    </div>
  );
}
