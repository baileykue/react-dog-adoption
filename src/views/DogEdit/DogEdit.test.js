import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogEdit from './DogEdit';

test('<DogEdit/> should render the specific dog form information', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/5/edit']}>
      <Route path="/dogs/:id/edit">
        <DogEdit />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('Dalmatian');
  expect(container).toMatchSnapshot();
});
