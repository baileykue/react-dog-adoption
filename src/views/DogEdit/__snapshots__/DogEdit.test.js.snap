// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DogEdit/> should render the specific dog form information 1`] = `
<div>
  <div>
    <div>
      <form
        class="form"
      >
        <label>
          Name: 
        </label>
        <input
          placeholder="Name"
          type="text"
          value="Stripe"
        />
        <label>
          Bio: 
        </label>
        <textarea
          placeholder="Bio"
          type="text"
        >
          This gal really loves chew toys and long walks on the beach. 
        </textarea>
        <label>
          Image: 
        </label>
        <input
          placeholder="Image"
          type="text"
          value="https://placedog.net/1000/563?id=135"
        />
        <label>
          Age: 
        </label>
        <input
          placeholder="Age"
          type="text"
          value="3"
        />
        <label>
          Breed: 
        </label>
        <input
          placeholder="Breed"
          type="text"
          value="Dalmatian"
        />
        <button>
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;
